public/js/length_visualisation.js

```diff
@@ -0,0 +1,339 @@
+import _ from 'underscore';
+import VisualisationHelpers from './visualisation_helpers';
+
+export default class LengthVisualisation {
+  constructor(query, svg_container, algorithm) {
+    this._svg_container_d3 =  d3.select(svg_container[0]);
+    this._svg = {};
+    this._query = query;
+    this._seq_type = this.get_seq_type(algorithm);
+    this._hits = query.hits;
+    this.visualisation_helpers = new VisualisationHelpers();
+
+    this._margin = {top: 30, right: 60, bottom: 70, left: 25};
+    this._width = 838 - this._margin.left - this._margin.right;
+    this._height = 400 - this._margin.top - this._margin.bottom;
+
+    this._data = this.fill_data();
+    this._x_scale = d3.scale.linear()
+        .domain([
+          0,
+          (d3.max([this._query.length,d3.max(this._data)]) * 1.05)
+        ])
+        .range([0, this._width]);
+    this._bins = this.create_bins();
+    this._data2 = this.fill_data2();
+    this._y_scale = d3.scale.linear()
```

Comment:
`this._y_scale` is used only once in `create_axis`. Why not define the variable in `create_axis` method?